src/connection/host_lookup.c

```diff
@@ -0,0 +1,674 @@
+static struct hostent *host_lookup_internal (const char *hostname, struct sockaddr *saddr, LOOKUP_TYPE lookup_type);
+
+/*
+ * hostent_alloc () - Allocate memory hostent structure.
+ * 
+ * return   : the hostent pointer.
+ *
+ */
+static struct hostent *
+hostent_alloc (char *ipaddr, char *hostname)
+{
+  struct hostent *hp;
+  char addr_trans_bi_buf[IPADDR_LEN];
+
+  if ((hp = (struct hostent *) malloc (sizeof (struct hostent))) == NULL)
+    {
+      return NULL;
+    }
+
+  hp->h_addrtype = AF_INET;
+  hp->h_length = IPv4_ADDR_LEN;
+
+  if (inet_pton (AF_INET, ipaddr, addr_trans_bi_buf) < 1)
+    {
+      fprintf (stderr, "Convertion IP address from text form to binary is failed");
+      FREE_MEM (hp);
+      return NULL;
+    }
+
+  hp->h_name = strdup (hostname);
+  hp->h_aliases = NULL;
+
+  if ((hp->h_addr_list = (char **) malloc (sizeof (char *) * HOSTNAME_LEN)) == NULL)
+    {
+      FREE_MEM (hp->h_name);
+      FREE_MEM (hp);
+      return NULL;
+    }
+
+  if ((hp->h_addr_list[0] = (char *) malloc (sizeof (char) * IPADDR_LEN)) == NULL)
+    {
+      FREE_MEM (hp->h_addr_list);
+      FREE_MEM (hp->h_name);
+      FREE_MEM (hp);
+
+      return NULL;
+    }
+
+  memcpy (hp->h_addr, addr_trans_bi_buf, 4);
+
+  return hp;
+}
+
+
+static struct hostent *
+host_lookup_internal (const char *hostname, struct sockaddr *saddr, LOOKUP_TYPE lookup_type)
+{
+  static struct hostent *hp;
+  int i, find_index = -1;
+
+  char addr_trans_ch_buf[IPADDR_LEN];
+  char addr_trans_bi_buf[IPADDR_LEN];
+  char hostname_buf[HOSTNAME_LEN];
+  char ipaddr_buf[IPADDR_LEN];
+  struct sockaddr_in *addr_trans = NULL;
+
+  if (hosts_conf_file_Load == LOAD_INIT)
+    {
+      if ((hosts_conf_file_Load = load_hosts_file ()) == LOAD_FAIL)
+	{
+	  return NULL;
+	}
+    }
+
+  addr_trans = (struct sockaddr_in *) saddr;
+
+  assert (((hostname != NULL) && (saddr == NULL)) || ((hostname == NULL) && (saddr != NULL)));
+
+  if (lookup_type == IPADDR_TO_HOSTNAME)
+    {
+      if (inet_ntop (AF_INET, &addr_trans->sin_addr, addr_trans_ch_buf, sizeof (addr_trans_ch_buf)) == NULL)
+	{
+	  fprintf (stderr, "Convertion IP address from binary form to text is failed");
+	  return NULL;
+	}
+
+    }
+
+  /*Look up in the user_host_Map */
+  if ((lookup_type == HOSTNAME_TO_IPADDR) && (user_host_Map.find (hostname) != user_host_Map.end ()))
+    {
+      hp = hostent_Cache[user_host_Map.find (hostname)->second];
+    }
+  else if ((lookup_type == IPADDR_TO_HOSTNAME) && (user_host_Map.find (addr_trans_ch_buf) != user_host_Map.end ()))
+    {
+      hp = hostent_Cache[user_host_Map.find (addr_trans_ch_buf)->second];
+    }
+  else
+    {
+      if (lookup_type == HOSTNAME_TO_IPADDR)
+	{
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_UHOST_CANT_LOOKUP_INFO, 1, hostname);
+	  fprintf (stdout, "%s\n", er_msg ());
+	}
+      else
+	{
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_UHOST_CANT_LOOKUP_INFO, 1, addr_trans_ch_buf);
+	  fprintf (stdout, "%s\n", er_msg ());
+	}
```

Comment:
This is the case of reverse lookup fails, it means we failed to find hostname by IP address. Because it is less important compared to IP address lookup, I recommend remove these lines.

Reply:
I got it. I will fix it.